source/common/common/logger.h

```diff
@@ -232,13 +244,18 @@ class Registry {
    */
   static spdlog::logger& getLog(Id id);
 
+  /**
+   * Init the singleton sink to use for all loggers.
+   *
+   * @param should_escape_ whether to escape c-style escape sequences in messages logged to the
+   * fallback stderr logger
+   */
+  static void initSink(bool should_escape_) { sink_ = DelegatingLogSink::init(should_escape_); }
```

Comment:
Yes, a few thoughts, though I haven't looked at this code in a long time. I agree of course about singletons in general, but I think it's not practical to require a context for logging (unless the code started that way) so there have to be some statics. The sink was also a lazy-init static prior to this PR (see the lazy-init getSink() in the 'before' view).

I don't see this as being largely worse than before, except the previous construct was thread-safe and now it is not. But in practice that shouldn't be an issue because initSink() is called during registry initialization. So I don't see a static-init fiasco here, @htuch -- it all looks properly lazy to me; the only theoretical problem is it's not thread-safe.

But here's my recommendation to improve this code; I don't remember why I didn't do it this way to begin with. Let's move the static sink to Logger::Context. Logger::Context is already a real class. It is properly instantiated on entry to MainContext and also to the testing framework. I think this is better than adding a static to Registry (which currently has no data). 

There's already a static instance pointer for Logger::Context, which is all we should need. It properly winds and unwinds itself too, so that MainCommon unit-tests can run inside the test framework.

Reply:
So I was actually able to revert back to the original API for the Registry. I just had to add the following to `Context::activate()`, not sure why I didn't think of this before:

```
Registry::getSink()->set_should_escape(should_escape_);
```

I tried following your approach @jmarantz, but then I'd have to change a lot of other files that call `Logger:Registry::getSink()` to `Logger::Context::getSink()`. I wasn't sure if this is what you meant, so I left it unchanged. Let me know if the current code is OK, since it does just keep the static variable contained in the function.